### PR TITLE
processing attribute #[path] of module 

### PR DESCRIPTION
**Pull request by andreevlex**
support two cases
- simple name file `foo.rs`
- declaration in mod.rs

#1211

### CODE REVIEW

**Comment by marcogroppo** [comment]
Awesome! In theory this should enable completions for `f32` and `f64`

File: crates/ra_hir/src/nameres/tests.rs [hunk 1]
```diff
@@ -364,6 +364,101 @@ fn module_resolution_works_for_raw_modules() {
     "###);
 }
 
+#[test]
+fn module_resolution_decl_path() {
```
Comment (matklad) [inline crates/ra_hir/src/nameres/tests.rs]: Lets move these and other tests that target specifically how `mod`s are resoved into a new `nameres/tests/mods.rs` submodule?

Comment (andreevlex) [reply crates/ra_hir/src/nameres/tests.rs]: Yes, of course

**Comment by matklad** [comment]
bors r+

Thanks!

**Comment by kjeremy** [comment]
On `rust-analyzer` we went from:

> Expressions of unknown type: 9603 (14%)
> Expressions of partially unknown type: 3788 (5%)

to

> Expressions of unknown type: 9599 (14%)
> Expressions of partially unknown type: 3802 (5%)

**Comment by marcogroppo** [comment]
I can confirm that RA can now display completions for `f32` and `f64` 🎉 Unfortunately they are incomplete, because we don't look for the `fXX_runtime` lang items in `libstd`.

**Comment by andreevlex** [comment]
@marcogroppo new issue?

**Comment by marcogroppo** [comment]
> @marcogroppo new issue?

I'll submit a PR!

**Pull request merged**